http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ProxyConfiguration.java

```diff
@@ -365,7 +365,11 @@ public void setNtlmWorkstation(String ntlmWorkstation) {
 
         @Override
         public Builder nonProxyHosts(Set<String> nonProxyHosts) {
-            this.nonProxyHosts = new HashSet<>(nonProxyHosts);
+            if (nonProxyHosts != null) {
+                this.nonProxyHosts = new HashSet<>(nonProxyHosts);
+            } else {
+                this.nonProxyHosts = Collections.emptySet();
```

Comment:
If we force the Null Value to be assigned as Empty set then it might break the below code

https://github.com/aws/aws-sdk-java-v2/blob/584ccb59e770177aeaa4c3b6bda4e24015b8ece9/http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ProxyConfiguration.java#L131-L145

Here we do a null check on nonProxyHosts , if user set the value of   nonProxyHosts as null internally we changing it to Collections.emptySet() might affect the getter ```public Set<String> nonProxyHosts() ```


Why don't we just do something s below 

```java
        @Override
        public Builder nonProxyHosts(Set<String> nonProxyHosts) {
            this.nonProxyHosts = nonProxyHosts != null ? new HashSet<>(nonProxyHosts) : null;
            return this;
        }
```